src/components/AnalyticsSearch/index.tsx

```diff
@@ -0,0 +1,367 @@
+                  );
+                }
+                if (tokenEntryKey === 'name' && !isAddress) {
+                  return token[tokenEntryKey].match(
+                    new RegExp(escapeRegExp(searchVal), 'i'),
+                  );
+                }
+                return false;
+              });
+              return regexMatches.some((m) => m);
+            })
+        : [];
+    return filtered;
+  }, [searchedTokens, searchVal]);
+
+  const filteredPairs = useMemo(() => {
+    const uniquePairs: any[] = [];
+    const pairsFound: any = {};
+    if (searchedPairs && searchedPairs.length > 0)
+      searchedPairs.map((pair) => {
+        if (!pairsFound[pair.id]) {
+          pairsFound[pair.id] = true;
+          uniquePairs.push(pair);
+        }
+        return true;
+      });
+
+    const filtered = uniquePairs
+      ? uniquePairs
+          .sort((pairA, pairB) => {
+            const pairAReserveETH = Number(pairA?.trackedReserveETH ?? 0);
+            const pairBReserveETH = Number(pairB?.trackedReserveETH ?? 0);
+            return pairAReserveETH > pairBReserveETH ? -1 : 1;
+          })
+          .filter((pair) => {
+            if (GlobalConst.blacklists.PAIR_BLACKLIST.includes(pair.id)) {
+              return false;
+            }
+            if (searchVal && searchVal.includes(' ')) {
+              const pairA = searchVal.split(' ')[0]?.toUpperCase();
+              const pairB = searchVal.split(' ')[1]?.toUpperCase();
+              return (
+                (pair.token0.symbol.includes(pairA) ||
+                  pair.token0.symbol.includes(pairB)) &&
+                (pair.token1.symbol.includes(pairA) ||
+                  pair.token1.symbol.includes(pairB))
+              );
+            }
+            if (searchVal && searchVal.includes('-')) {
+              const pairA = searchVal.split('-')[0]?.toUpperCase();
+              const pairB = searchVal.split('-')[1]?.toUpperCase();
+              return (
+                (pair.token0.symbol.includes(pairA) ||
+                  pair.token0.symbol.includes(pairB)) &&
+                (pair.token1.symbol.includes(pairA) ||
+                  pair.token1.symbol.includes(pairB))
+              );
+            }
+            const regexMatches = Object.keys(pair).map((field) => {
+              const isAddress = searchVal.slice(0, 2) === '0x';
+              if (field === 'id' && isAddress) {
+                return pair[field].match(
+                  new RegExp(escapeRegExp(searchVal), 'i'),
+                );
+              }
+              if (field === 'token0') {
+                return (
+                  pair[field].symbol.match(
+                    new RegExp(escapeRegExp(searchVal), 'i'),
+                  ) ||
+                  pair[field].name.match(
+                    new RegExp(escapeRegExp(searchVal), 'i'),
+                  )
+                );
+              }
+              if (field === 'token1') {
```

Comment:
What's the different between this and the if above, why not combine them into 1?

Reply:
Fixed this and pushed another commit